### PR TITLE
fix(proxy): cap client abort drain window

### PR DESCRIPTION
**Pull request by Brisbanehuang**
## Summary

This caps the background drain window after a downstream client aborts at 60 seconds, even when the provider's streaming idle timeout is configured to a larger value.

The normal streaming idle timeout still controls live client connections. This change only applies after the client has already disconnected and the proxy is doing best-effort background draining so it can still capture terminal markers / usage and finalize the request correctly.

## Why

#1251 added a bounded drain fallback when `streamingIdleTimeoutMs` is disabled (`0` / `Infinity`). However, when a provider sets a long idle timeout such as 120s or 600s, the client-abort drain window currently inherits that full duration. That can keep an upstream stream, accumulated response chunks, session state, and abort listeners alive long after there is no downstream client left to receive the response.

A fixed 60s cap keeps the drain behavior best-effort while giving resource retention a predictable upper bound.

## Trade-off

This is intentionally a behavior change for client-aborted streams. If an upstream remains silent for more than 60 seconds after the downstream client has already disconnected and only then emits final usage / terminal markers, the proxy may stop draining before capturing that final accounting data.

That trade-off is limited to already-disconnected clients. Active client streams continue to use the provider's configured `streamingIdleTimeoutMs`.

## Validation

- `bun run test tests/unit/proxy/response-handler-client-abort-drain.test.ts`
- `bunx biome check src/app/v1/_lib/proxy/response-handler.ts tests/unit/proxy/response-handler-client-abort-drain.test.ts`
- `bun run typecheck`

<details open><summary><h3>Greptile Summary</h3></summary>

This PR caps the background drain window to a fixed 60 seconds after a downstream client disconnects, regardless of the provider's configured `streamingIdleTimeoutMs`. It also moves `startIdleTimer`/`clearIdleTimer` and the `chunks` array to the outer scope so they can be used from the client-abort listener, and fixes a bug where the idle-timeout finalization path incorrectly reported `STREAM_IDLE_TIMEOUT` instead of `CLIENT_ABORTED` when the client had already disconnected before the idle timer fired.

- `clientAbortDrainTimeoutMs` is now always `CLIENT_ABORT_DRAIN_MAX_MS` (60 s) instead of inheriting the provider's potentially large `streamingIdleTimeoutMs`.
- When the client aborts and no idle timer is already running (pre-body hang), `startIdleTimer()` is called immediately so a short `streamingIdleTimeoutMs` can provide earlier termination than the 60 s cap.
- The `isIdleTimeout` finalization branch now passes the actual `clientAborted` flag and uses `"CLIENT_ABORTED"` as the abort reason when appropriate, fixing the mislabelled accounting for client-aborted streams that happen to be terminated by the idle timer.
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge. The change only affects already-disconnected clients; active streams are untouched.

The diff is targeted and well-tested. The 60 s cap only applies after client disconnect; active streams are unaffected. The clientAborted flag fix corrects a mislabel, and all four key edge cases are covered by new tests.

No files require special attention.
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src/app/v1/_lib/proxy/response-handler.ts | Caps drain window to 60 s, lifts idle-timer helpers to outer scope, and fixes CLIENT_ABORTED vs STREAM_IDLE_TIMEOUT mislabelling; logic is sound and all four distinct abort paths are handled correctly. |
| tests/unit/proxy/response-handler-client-abort-drain.test.ts | Adds four targeted test cases (pre-body hang, active-chunks drain independence, existing-idle-timer preservation, and 60 s cap) that together cover the key edge cases introduced by this change. |

</details>

</details>

<sub>Reviews (6): Last reviewed commit: ["fix(proxy): keep abort-drain idle as cli..."](https://github.com/ding113/claude-code-hub/commit/b2aeba0af2b0f61c68d1737e5e3f23421a80a420) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=37257152)</sub>

### CODE REVIEW

File: src/app/v1/_lib/proxy/response-handler.ts [hunk 1]
```diff
@@ -2322,7 +2324,7 @@ export class ProxyResponseHandler {
     const abortController = new AbortController();
     const idleTimeoutMs =
       provider.streamingIdleTimeoutMs > 0 ? provider.streamingIdleTimeoutMs : Infinity;
-    const clientAbortDrainTimeoutMs = idleTimeoutMs === Infinity ? 60_000 : idleTimeoutMs;
+    const clientAbortDrainTimeoutMs = CLIENT_ABORT_DRAIN_MAX_MS;
```
[the remark this answers is not in the document]
Comment (Brisbanehuang) [reply src/app/v1/_lib/proxy/response-handler.ts:2327]: Handled with a slightly different split: the total client-abort drain cap stays fixed at 60s, and client abort now also arms the existing idle watchdog. That preserves the 60s total drain window for active streams while still letting a short `streamingIdleTimeoutMs` release pre-body hangs before the first chunk arrives. Added coverage for both cases: active chunks with a 5s idle timeout are allowed to drain until the 60s cap, and pre-body hangs with a 5s idle timeout are aborted by the idle watchdog.

File: src/app/v1/_lib/proxy/response-handler.ts [hunk 2]
```diff
@@ -2344,6 +2405,9 @@ export class ProxyResponseHandler {
       // still drain buffered final usage and record the request as successful.
       // Idle/response timeout paths still abort via abortController.
       clearClientAbortDrainTimer();
+      if (!idleTimeoutId) {
+        startIdleTimer();
```
[the remark this answers is not in the document]
Comment (Brisbanehuang) [reply src/app/v1/_lib/proxy/response-handler.ts:2409]: Fixed in b2aeba0. The idle-timeout catch path now preserves clientAborted when the idle watchdog fires during a post-abort drain, so those records continue through the 499 CLIENT_ABORTED finalization path while non-client-aborted idle timeouts remain STREAM_IDLE_TIMEOUT.
